akka-remote/src/main/scala/akka/remote/artery/ArterySettings.scala

```diff
@@ -20,6 +20,7 @@ import akka.util.Helpers.Requiring
 import akka.util.Helpers.toRootLowerCase
 import akka.util.WildcardIndex
 import akka.util.ccompat.JavaConverters._
+import sun.net.util.IPAddressUtil
```

Comment:
We should avoid using `sun.net` stuff as it's not an official/public part of the JDK (and can be removed whenever in future JDK versions).  

In `akka.io.dns.internal.AsyncDnsResolver#ipv6Address` there is a regexp, let's create a `private[akka]` `isIPv6Address(String): Boolean` method there and call from here instead.

Reply:
That's really a complex regexp😂